src/server/apsystem.ts

```diff
@@ -185,14 +185,16 @@ export default class ActivityPubSystem {
   }
 
   async getActor (actorURL: string): Promise<APActor> {
-  // resolve actor data with fetch
-    const response = await this.fetch(actorURL, {
+  // resolve actor data with signedFetch
+    const response = await this.signedFetch(this.publicURL, {
```

Comment:
We cannot use the publicURL as an actor, we need to modify the method to pass in a `fromActor` parameter which represents the actor we're fetching this data on behalf of.